### PR TITLE
Add slip44 to coin definition and codegen blockchain type

### PR DESCRIPTION
**Pull request by hewigovens**
This was intended for #486, we explicitly add `slip44` to coin definition, so we can support derivation path that doesn't contain coin type.

### CODE REVIEW

File: src/HDWallet.h [hunk 1]
```diff
@@ -59,7 +59,7 @@ class HDWallet {
     virtual ~HDWallet();
 
     /// Returns the private key at the given derivation path.
-    PrivateKey getKey(const DerivationPath& derivationPath) const;
+    PrivateKey getKey(enum TWCoinType coin, const DerivationPath& derivationPath) const;
```
Comment (vikmeup) [inline src/HDWallet.h]: it's a little bit not intuitive to provide coin as slip44 and derivation path at the same time, is there a way to init derivationPath with a coin type?

Comment (hewigovens) [reply src/HDWallet.h]: we need coin to get the curve type

Comment (vikmeup) [reply src/HDWallet.h]: but `derivationPath` has a coin right? I would personally keep it as simple as possible, not a big fan of slip048 standard, should we just follow bip44 for all coins?

Comment (hewigovens) [reply src/HDWallet.h]: Yes, we can close this

**Pull request closed without merging**